### PR TITLE
Add SongsTableTest

### PR DESCRIPTION
**Pull request by adityahemanth**

### CODE REVIEW

File: app/src/test/java/com/bhakti/db/tables/SongsTableTest.java [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+  public void insertOrUpdate_withASingleValidEntry() throws Exception {
+    songsTable.insertOrUpdate(123, FAKE_SONG_INFO_1).get();
+
+    SongInfo retrievedSong = songsTable.get(123).get();
+    compareSongInfos(retrievedSong, FAKE_SONG_INFO_1);
+  }
+
+  @Test
+  public void insertOrUpdate_withNullEntry() throws Exception {
+    songsTable.insertOrUpdate(1, SongInfo.builder().build()).get();
```
Comment (sasi306) [inline app/src/test/java/com/bhakti/db/tables/SongsTableTest.java]: add argument tags.

File: app/src/test/java/com/bhakti/db/tables/SongsTableTest.java [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+    testActivity = Robolectric.buildActivity(Activity.class)
+        .create()
+        .resume()
+        .get();
+
+    songsTable = new SongsTable(testActivity.getApplicationContext());
+  }
+
+  @Test
+  public void insertOrUpdate_withASingleValidEntry() throws Exception {
```
Comment (sasi306) [inline app/src/test/java/com/bhakti/db/tables/SongsTableTest.java]: do you want to add a case for replacing a song.

File: app/src/main/java/com/bhakti/db/tables/SongsTable.java [hunk 3]
```diff
@@ -133,6 +135,31 @@ public void onUpgrade(SQLiteDatabase sqLiteDatabase, int i, int i1) {
...
+          ContentValues values = buildContentValues(entry.getValue());
+          values.put(SongEntry._ID, entry.getKey());
+          db.insertWithOnConflict(SongEntry.TABLE_NAME, null, values,
+              SQLiteDatabase.CONFLICT_REPLACE);
+        }
+        db.setTransactionSuccessful();
+      } finally {
+        db.endTransaction();
+      }
+      return null;
```
[the remark this answers is not in the document]
Comment (snmlabs) [reply app/src/main/java/com/bhakti/db/tables/SongsTable.java]: Yes, we need this for lambdas

File: app/src/test/java/com/bhakti/db/tables/AndroidManifest.xml [hunk 4]
```diff
@@ -0,0 +1,8 @@
+<manifest
```
[the remark this answers is not in the document]
Comment (snmlabs) [reply app/src/test/java/com/bhakti/db/tables/AndroidManifest.xml]: Yes, we need separate manifest for each test module.

File: app/src/test/java/com/bhakti/db/tables/SongsTableTest.java (hunk 2 above)
Comment (snmlabs) [reply app/src/test/java/com/bhakti/db/tables/SongsTableTest.java]: done.

File: app/src/test/java/com/bhakti/db/tables/SongsTableTest.java (hunk 1 above)
Comment (snmlabs) [reply app/src/test/java/com/bhakti/db/tables/SongsTableTest.java]: done.

**Pull request merged**